src/main/java/software/amazon/event/kafkaconnector/offloading/S3EventBridgeEventDetailValueOffloading.java

```diff
@@ -0,0 +1,225 @@
+import software.amazon.event.kafkaconnector.EventBridgeResult;
+import software.amazon.event.kafkaconnector.cache.Cache;
+import software.amazon.event.kafkaconnector.cache.FifoCache;
+import software.amazon.event.kafkaconnector.cache.MessageDigestCacheKey;
+import software.amazon.event.kafkaconnector.logging.ContextAwareLoggerFactory;
+import software.amazon.event.kafkaconnector.mapping.EventBridgeMappingResult;
+import software.amazon.event.kafkaconnector.util.MappedSinkRecord;
+import software.amazon.event.kafkaconnector.util.ThrowingFunction;
+import software.amazon.event.kafkaconnector.util.ThrowingFunctionApplyException;
+
+public class S3EventBridgeEventDetailValueOffloading
+    implements EventBridgeEventDetailValueOffloadingStrategy {
+
+  public static final String JSON_PATH_PREFIX = "$.detail.value";
+
+  private static final int SDK_TIMEOUT = 5000; // timeout in milliseconds for SDK calls
+
+  // maximum size of cache (by Map.Entry<MessageDigestCacheKey, UUID>) => ~16MiB
+  // size of each Map.Entry<MessageDigestCacheKey, UUID> is 168Byte (for x86_64)
+  private static final int MAX_CACHE_SIZE = 100_000;
+
+  private static final Logger logger =
+      ContextAwareLoggerFactory.getLogger(S3EventBridgeEventDetailValueOffloading.class);
+
+  private static final Configuration jsonPathConfiguration =
+      defaultConfiguration()
+          .addOptions(
+              SUPPRESS_EXCEPTIONS // suppress exception otherwise
+              // com.jayway.jsonpath.ReadContext#read throws an exception if JSON path could not
+              // be found
+              );
+
+  private final String bucketName;
+  private final S3AsyncClient client;
+
+  private final Cache<MessageDigestCacheKey, UUID> s3ObjectKeyCache;
+  private final Supplier<UUID> idGenerator;
+
+  private static final JsonPath jsonPathAdd = JsonPath.compile("$");
+
+  private final String jsonPathExp;
+  private final JsonPath jsonPathRemove;
+
+  public S3EventBridgeEventDetailValueOffloading(
+      final S3AsyncClient client,
+      final String bucketName,
+      final String jsonPathExp,
+      final Cache<MessageDigestCacheKey, UUID> s3ObjectKeyCache,
+      final Supplier<UUID> idGenerator) {
+
+    this.bucketName = bucketName;
+    this.client = client;
+
+    this.s3ObjectKeyCache = s3ObjectKeyCache;
+    this.idGenerator = idGenerator;
+
+    this.jsonPathExp = jsonPathExp;
+    this.jsonPathRemove = transformToPutEventsRequestEntry(jsonPathExp);
+  }
+
+  public S3EventBridgeEventDetailValueOffloading(
+      final S3AsyncClient client, final String bucketName, final String jsonPathExp) {
+
+    this.bucketName = bucketName;
+    this.client = client;
+
+    this.s3ObjectKeyCache = new FifoCache<>(MAX_CACHE_SIZE);
+    this.idGenerator = UUID::randomUUID;
+
+    this.jsonPathExp = jsonPathExp;
+    this.jsonPathRemove = transformToPutEventsRequestEntry(jsonPathExp);
+  }
+
+  private static JsonPath transformToPutEventsRequestEntry(String value) {
```

Comment:
nit: is this signature correct? seems misaligned with what it's doing

Reply:
Yes, my fault. It's fixed.